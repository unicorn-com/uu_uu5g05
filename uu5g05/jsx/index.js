// used from libraries depending on uu5g05 (this gets bundled into them); used due to uu5devkitg01-plugin's .babelrc config regarding @babel/preset-react preset
const Uu5 = require("uu5g05");
// the fallback is for temporary flow as described in ./jsx-runtime.js
Object.assign(exports, Uu5._jsx || { createElement: Uu5.Utils.Element.create });
