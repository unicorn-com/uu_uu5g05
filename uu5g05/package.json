{
  "name": "uu5g05",
  "version": "1.40.1",
  "description": "uu5 Library contains basic hooks, providers, components, utils and environment for easy development of uuApps or uuLibs.",
  "license": "License Commercial",
  "namespace": "Uu5",
  "spec": "uu5-lib; uu_appg01_devkit@6.6.0; uu5devkitg01-plugin@1.5.1",
  "dependencies": {
    "prop-types": "15.7.2",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "regenerator-runtime": "0.13.7",
    "uu5stringg01": "^1.10.0"
  },
  "uuBuildSettings": {
    "externals": {
      "react": true,
      "react-dom": true,
      "prop-types": true
    }
  },
  "devDependencies": {
    "@emotion/css": "11.9.0",
    "@mdi/font": "7.0.96",
    "memoize-one": "5.1.1",
    "systemjs": "0.19.47",
    "uu_appg01_devkit": "^6.6.0",
    "uu_appg01_oidc": "^4.0.0",
    "uu_datatypesg01": "^0.1.1",
    "uu5devkitg01-plugin": "^1.5.1",
    "uu5g05-test": "*",
    "uu5loaderg01": "^1.7.0"
  },
  "engines": {
    "node": ">=12.0.0"
  },
  "scripts": {
    "clean": "uu_appg01_devkit clean",
    "test": "uu_appg01_devkit test",
    "build": "node tools/prebuild.js && uu_appg01_devkit build",
    "package": "uu_appg01_devkit package",
    "dist": "uu_appg01_devkit package",
    "start": "uu_appg01_devkit start",
    "prettify": "uu_appg01_devkit prettify",
    "vulnerabilityScan": "uu_appg01_devkit vulnerabilityScan",
    "uuAppDevKitUpgrade": "uu_appg01_devkit uuAppDevKitUpgrade",
    "uu5Generate": "uu5devkitg01-plugin generate",
    "uu5LsiExport": "uu5devkitg01-plugin lsiExport",
    "uu5LsiImport": "uu5devkitg01-plugin lsiImport",
    "uu5DocUpdateExamples": "uu5devkitg01-plugin docUpdateExamples"
  },
  "main": "src/index.js",
  "files": [
    "test/setup",
    "dist",
    "dist-node",
    "jsx"
  ],
  "prettier": {
    "printWidth": 120
  },
  "jest": {
    "transformIgnorePatterns": [
      "^(?!.*/node_modules/uu_appg01).*/node_modules/",
      "uu5g05-test"
    ]
  }
}
