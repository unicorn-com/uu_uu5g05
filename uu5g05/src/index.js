import "./_internal/polyfills/polyfills.js";

import * as Exports from "./exports.js";
import LibraryRegistry from "./utils/library-registry.js";
import "./_internal/startup.js";

export * from "./exports.js";
export default Exports;

if (process.env.NODE_ENV !== "test") {
  console.log(
    `${process.env.NAME}-${process.env.VERSION} Â© Unicorn\nTerms of Use: https://unicorn.com/tou/${process.env.NAME}`,
  );
}
LibraryRegistry.registerLibrary({
  name: process.env.NAME,
  version: process.env.VERSION,
  namespace: process.env.NAMESPACE,
});

// we'll still export uu5g05 into global variable so that JSX pragmas can be used
// (Uu5g05.Utils.Element.create + Uu5g05.Fragment), however, we'll prevent from adding
// anything to the value
// eslint-disable-next-line no-import-assign
const frozenExports = Object.freeze(Exports);
const publish = (scope, exports) => {
  scope.Uu5 = exports;
  scope.Uu5g05 = {
    Utils: {
      Element: {
        create: exports.Utils.Element.create,
      },
    },
  };
  Object.defineProperty(scope.Uu5g05, "Fragment", Object.getOwnPropertyDescriptor(exports, "Fragment"));
};
if (typeof window !== "undefined") publish(window, frozenExports);
if (process.env.NODE_ENV === "test" && typeof global !== "undefined") publish(global, frozenExports); // Jest tests
