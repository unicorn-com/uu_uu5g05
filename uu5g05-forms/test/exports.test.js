import { Utils } from "uu5g05";
import Uu5Forms from "uu5g05-forms";

const memo = Utils.Component.memo(() => {});
describe(`Uu5Forms`, () => {
  it("exported API", async () => {
    let apiMap = {};
    addKeys(Uu5Forms, apiMap);
    expect(apiMap).toMatchSnapshot();
  });
});

function addKeys(obj, result, filter = (key, obj, prefix) => true, prefix = "", addedSet = new Set()) {
  if (prefix) result[prefix] = typeof obj;
  if (!obj || (typeof obj !== "object" && typeof obj !== "function")) return;
  if (addedSet.has(obj)) return;
  addedSet.add(obj);
  for (let k of Object.getOwnPropertyNames(obj)) {
    if (/^_/.test(k)) continue;
    if (
      /^(defaultProps|displayName|isStateless|isUu5PureComponent|length|logger|name|nestingLevel|propTypes|prototype|uu5ComponentType|uu5Tag|getDerivedStateFromError|getDerivedStateFromProps)$/.test(
        k,
      )
    ) {
      continue;
    }
    if (obj.$$typeof === memo.$$typeof && k in memo) continue;
    if (!filter(k, obj, prefix)) continue;
    let childFilter;
    if (/^[a-z]/.test(k) || /^[A-Z_0-9]+$/.test(k)) childFilter = () => false;
    else if (/^[A-Z]/.test(k)) childFilter = (k, obj) => k in obj;
    else continue;
    addKeys(obj[k], result, childFilter, prefix ? prefix + "." + k : k, addedSet);
  }
}
