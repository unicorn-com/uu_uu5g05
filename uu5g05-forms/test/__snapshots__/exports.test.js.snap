// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uu5Forms exported API 1`] = `
{
  "CancelButton": "function",
  "Checkbox": "function",
  "Checkbox.Input": "function",
  "Checkbox.RadioInput": "function",
  "Checkbox.validationMap": "object",
  "Checkboxes": "function",
  "Checkboxes.Input": "function",
  "Checkboxes.Input.validationMap": "object",
  "Color": "function",
  "Color.Input": "function",
  "Color.Input.validationMap": "object",
  "Color.Picker": "function",
  "Color.validationMap": "object",
  "Date": "function",
  "Date.Box": "function",
  "Date.Input": "function",
  "Date.Input.Box": "function",
  "Date.Input.validationMap": "object",
  "Date.validationMap": "object",
  "DateRange": "function",
  "DateRange.Box": "function",
  "DateRange.Input": "function",
  "DateRange.Input.Box": "function",
  "DateRange.Input.validationMap": "object",
  "DateRange.validationMap": "object",
  "DateTime": "function",
  "DateTime.Input": "function",
  "DateTime.Input.validationMap": "object",
  "DateTime.validationMap": "object",
  "DateTimeRangeProvider": "function",
  "Email": "function",
  "Email.Input": "function",
  "Email.Input.validationMap": "object",
  "File": "function",
  "File.Input": "function",
  "File.Input.validationMap": "object",
  "File.validationMap": "object",
  "Form": "function",
  "Form.Provider": "function",
  "Form.Provider.State": "function",
  "Form.Provider.validationMap": "object",
  "Form.UnhandledError": "function",
  "Form.View": "function",
  "Form.View.COLUMN_GAP_SPACING": "string",
  "Form.View.ROW_GAP_SPACING": "string",
  "FormCheckbox": "function",
  "FormCheckbox.Input": "function",
  "FormCheckbox.RadioInput": "function",
  "FormCheckbox.validationMap": "object",
  "FormCheckboxes": "function",
  "FormCheckboxes.Input": "function",
  "FormColor": "function",
  "FormColor.Input": "function",
  "FormColor.Picker": "function",
  "FormColor.validationMap": "object",
  "FormDate": "function",
  "FormDate.Box": "function",
  "FormDate.Input": "function",
  "FormDate.validationMap": "object",
  "FormDateRange": "function",
  "FormDateRange.Box": "function",
  "FormDateRange.Input": "function",
  "FormDateRange.validationMap": "object",
  "FormDateTime": "function",
  "FormDateTime.Input": "function",
  "FormDateTime.validationMap": "object",
  "FormEmail": "function",
  "FormEmail.Input": "function",
  "FormFile": "function",
  "FormFile.Input": "function",
  "FormFile.validationMap": "object",
  "FormIcon": "function",
  "FormIcon.Input": "function",
  "FormIcon.Picker": "function",
  "FormIcon.validationMap": "object",
  "FormLink": "function",
  "FormLink.Input": "function",
  "FormLsi": "function",
  "FormLsi.Input": "function",
  "FormLsi.validationMap": "object",
  "FormMonth": "function",
  "FormMonth.Input": "function",
  "FormMonth.validationMap": "object",
  "FormMonthRange": "function",
  "FormMonthRange.Box": "function",
  "FormMonthRange.Input": "function",
  "FormMonthRange.validationMap": "object",
  "FormNumber": "function",
  "FormNumber.Input": "function",
  "FormPassword": "function",
  "FormPassword.Input": "function",
  "FormPassword.validationMap": "object",
  "FormQuarter": "function",
  "FormQuarter.Box": "function",
  "FormQuarter.Input": "function",
  "FormQuarter.validationMap": "object",
  "FormQuarterRange": "function",
  "FormQuarterRange.Box": "function",
  "FormQuarterRange.Input": "function",
  "FormQuarterRange.validationMap": "object",
  "FormRadios": "function",
  "FormRadios.Input": "function",
  "FormSelect": "function",
  "FormSelect.Field": "function",
  "FormSelect.Input": "function",
  "FormSelect.Options": "function",
  "FormSelect.SelectedOptions": "function",
  "FormSlider": "function",
  "FormSwitchSelect": "function",
  "FormSwitchSelect.Input": "function",
  "FormText": "function",
  "FormText.Input": "function",
  "FormTextArea": "function",
  "FormTextArea.Input": "function",
  "FormTextSelect": "function",
  "FormTextSelect.Field": "function",
  "FormTextSelect.Input": "function",
  "FormTextSelect.Options": "function",
  "FormTextSelect.SelectedOptions": "function",
  "FormTextSelectAsync": "function",
  "FormTextSelectAsync.Field": "function",
  "FormTextSelectAsync.Input": "function",
  "FormTextSelectAsync.Options": "function",
  "FormTextSelectAsync.SelectedOptions": "function",
  "FormTextSelectAsync.validationMap": "object",
  "FormTime": "function",
  "FormTime.Input": "function",
  "FormTime.validationMap": "object",
  "FormWeek": "function",
  "FormWeek.Input": "function",
  "FormWeek.validationMap": "object",
  "FormWeekRange": "function",
  "FormWeekRange.Box": "function",
  "FormWeekRange.Input": "function",
  "FormWeekRange.validationMap": "object",
  "FormYear": "function",
  "FormYear.Input": "function",
  "FormYear.validationMap": "object",
  "FormYearRange": "function",
  "FormYearRange.Box": "function",
  "FormYearRange.Input": "function",
  "FormYearRange.validationMap": "object",
  "Icon": "function",
  "Icon.Input": "function",
  "Icon.Input.validationMap": "object",
  "Icon.Picker": "function",
  "Icon.validationMap": "object",
  "Label": "function",
  "Link": "function",
  "Link.Input": "function",
  "Link.Input.validationMap": "object",
  "Lsi": "function",
  "Lsi.Input": "function",
  "Lsi.Input.validationMap": "object",
  "Lsi.validationMap": "object",
  "Message": "function",
  "Month": "function",
  "Month.Input": "function",
  "Month.Input.validationMap": "object",
  "Month.validationMap": "object",
  "MonthRange": "function",
  "MonthRange.Box": "function",
  "MonthRange.Input": "function",
  "MonthRange.Input.Box": "function",
  "MonthRange.Input.validationMap": "object",
  "MonthRange.validationMap": "object",
  "Number": "function",
  "Number.Input": "function",
  "Number.Input.validationMap": "object",
  "Password": "function",
  "Password.Input": "function",
  "Password.Input.validationMap": "object",
  "Password.validationMap": "object",
  "Quarter": "function",
  "Quarter.Box": "function",
  "Quarter.Input": "function",
  "Quarter.Input.Box": "function",
  "Quarter.Input.validationMap": "object",
  "Quarter.validationMap": "object",
  "QuarterRange": "function",
  "QuarterRange.Box": "function",
  "QuarterRange.Input": "function",
  "QuarterRange.Input.Box": "function",
  "QuarterRange.Input.validationMap": "object",
  "QuarterRange.validationMap": "object",
  "Radios": "function",
  "Radios.Input": "function",
  "Radios.Input.validationMap": "object",
  "ResetButton": "function",
  "Select": "function",
  "Select.Field": "function",
  "Select.Input": "function",
  "Select.Options": "function",
  "Select.SelectedOptions": "function",
  "Slider": "function",
  "Slider.Input": "function",
  "Slider.Input.validationMap": "object",
  "Slider.validationMap": "object",
  "SubmitButton": "function",
  "SwitchSelect": "function",
  "SwitchSelect.Input": "function",
  "SwitchSelect.Input.validationMap": "object",
  "Text": "function",
  "Text.Input": "function",
  "Text.Input.validationMap": "object",
  "TextArea": "function",
  "TextArea.Input": "function",
  "TextArea.Input.validationMap": "object",
  "TextSelect": "function",
  "TextSelect.Field": "function",
  "TextSelect.Input": "function",
  "TextSelect.Options": "function",
  "TextSelect.SelectedOptions": "function",
  "TextSelectAsync": "function",
  "TextSelectAsync.Field": "function",
  "TextSelectAsync.Input": "function",
  "TextSelectAsync.Input.validationMap": "object",
  "TextSelectAsync.Options": "function",
  "TextSelectAsync.SelectedOptions": "function",
  "TextSelectAsync.validationMap": "object",
  "Time": "function",
  "Time.Input": "function",
  "Time.Input.validationMap": "object",
  "Time.validationMap": "object",
  "Week": "function",
  "Week.Input": "function",
  "Week.Input.validationMap": "object",
  "Week.validationMap": "object",
  "WeekRange": "function",
  "WeekRange.Box": "function",
  "WeekRange.Input": "function",
  "WeekRange.Input.Box": "function",
  "WeekRange.Input.validationMap": "object",
  "WeekRange.validationMap": "object",
  "Wizard": "object",
  "Wizard.Content": "function",
  "Wizard.PreviousButton": "function",
  "Wizard.Provider": "function",
  "Wizard.Stepper": "function",
  "Wizard.SubmitButton": "function",
  "Year": "function",
  "Year.Input": "function",
  "Year.Input.validationMap": "object",
  "Year.validationMap": "object",
  "YearRange": "function",
  "YearRange.Box": "function",
  "YearRange.Input": "function",
  "YearRange.Input.Box": "function",
  "YearRange.Input.validationMap": "object",
  "YearRange.validationMap": "object",
  "useDateTimeFormat": "function",
  "useForm": "function",
  "useFormApi": "function",
  "useValidatorMap": "function",
  "useWizard": "function",
  "withExtensionInput": "function",
  "withFormInput": "function",
  "withFormItem": "function",
  "withSubformItem": "function",
  "withValidationInput": "function",
  "withValidationMap": "function",
}
`;
